Created `.targets` file to copy .sql files to the output directory

Ts-Pytham opened the pull request.

Ts-Pytham committed:
MsBuild task and CopySqlFiles created

Commit:
Change in access modifier of the CopySqlFiles class and change in YeSql.Net.csproj

Commit:
Added AfterTargents Type "Build"

DevD4v3 committed:
changes merge

DevD4v3 committed:
Added a .targets file to copy .sql files to the output directory

DevD4v3 committed:
Added comment in .csproj

DevD4v3 commented:
The problem encountered is that when using the following command:
```sh
dotnet publish -o out
```
.sql files are not copied to the `out` directory when publishing the project.

DevD4v3 committed:
Added a target to copy the .sql files to the publish directory

DevD4v3 committed:
Added `SkipUnchangedFiles` parameter in `Copy` task

This causes the Copy task to not copy the source files if no changes have been made.

The Copy task considers files (source and destination) to be unchanged if they have the same size and the same last modified time.

DevD4v3 commented:
The previous problem was solved, but another problem occurred: A `bin` folder is created unnecessarily.

### Samples
![image](https://github.com/ose-net/yesql.net/assets/43916038/a83628ad-f204-4975-9953-676a29729fde)

![image](https://github.com/ose-net/yesql.net/assets/43916038/87ef905d-9e68-4906-b2ab-8dddc273596d)

DevD4v3 committed:
Added `Exclude` parameter to omit folders like ./bin, ./obj, etc.

See https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#defaultitemexcludes

DevD4v3 committed:
Removed backslash

DevD4v3 committed:
Added logs a message during a build or publish

DevD4v3 committed:
Exclude default publish folders

DevD4v3 committed:
updated .csproj

DevD4v3 committed:
Move `build` folder out of the project directory

DevD4v3 commented:
Previous problems have been solved, however, the library may fail intentionally through a client error.
For example, imagine that the project structure of a client is:
```sh
└── ConsoleApp/
    ├── bin/
    ├── obj/
    ├── sql/
    │   └── sample.sql
    ├── ConsoleApp.csproj
    ├── Program.cs
    └── sample.sql
```
A client can execute the following command:
```sh
dotnet publish -o webpublish
```
A directory named `webpublish` is created in `/ConsoleApp` and this directory will have a `YeSQL` folder with .sql files.

When the client executes the `dotnet build` command, the `webpublish` folder will also be copied to `/bin/Debug/net7.0/YeSQL`:
![image](https://github.com/ose-net/yesql.net/assets/43916038/278940e1-326e-4dfa-a6a7-741f3e157ac7)

This is normal behavior, it cannot be avoided, however, this causes the library to load the same .sql files, so it leads to an exception for having duplicate tags.
And also the library will start searching for .sql files from the `YeSQL` folder when the `YeSqlLoader.Load()` method is used.

## Alternatives
To avoid this exception, the client has several solutions:
- Create the publish directory outside the **project folder**. Absolute paths can be used in this option:
```sh
dotnet publish -o C:\Program Files\MySqlFiles
```
- Execute the `dotnet publish` command without specifying an output folder, so the publish directory would be created in `/bin/Debug/net7.0`.
- Attach a publish directory that starts with a period (`.`) as *.webpublish*:
```sh
dotnet publish -o .webpublish
```
- Redefine the `DefaultItemExcludes` property from the .csproj file to exclude the `webpublish` folder when the project is built with `dotnet build`:
```xml
<PropertyGroup>
   <DefaultItemExcludes>
     $(DefaultItemExcludes);webpublish/**
   </DefaultItemExcludes>
</PropertyGroup>
```
The one that is recommended is not to create the publish directory inside the **project folder**, after all git would track the changes and it should not be like that. If either option is followed, the library does not generate any errors.

This is not really a library problem but should be documented in a section of README.md for the client to take precautions.

DevD4v3 committed:
Added comments to `.targets` file

DevD4v3 committed:
Removed `DefaultExcludesInProjectFolder` property

This property is not required.

DevD4v3 committed:
Revert "Removed `DefaultExcludesInProjectFolder` property"

This reverts commit 188e8d3f6d1b1dc49879ce572c71f4dd5ec13346.

DevD4v3 committed:
Updated comments

DevD4v3 commented:
This PR should be tested with .NET Framework 4.6+ using the latest version of MSBuild (17.0).
